add a sub-section about optim_m1qn3

mjlosch opened the pull request:
## What changes does this PR introduce?
docs update
A new sub-section about optim_m1qn3 briefly describes alternative optimization code in
a non-MITgcm repository. Therefore, this description mostly refer to the instructions on that site.

## Does this PR introduce a breaking change? 
No

## Suggested addition to `tag-index`
add a sub-section about optim_m1qn3

christophernhill commented:
@gaelforget GitHub suggested you might take a look at this. If. you have any comments let us know.

christophernhill commented:
@mjlosch and @hongandyan how is this optim setup related to the latest ECCO pieces?

hongandyan commented:
We've been using M1QN3 for LLC270 optimization for a while (suggested by Matt),
though it's an old check out from Martin's CVS.

gaelforget commented:
@owang01 should confirm or otherwise but I believe that ECCO production is currently using 
[](http://wwwcvs.mitgcm.org/viewvc/MITgcm/MITgcm_contrib/ecco_utils/ecco_v4_release3_optimization/)

mjlosch commented:
As far as I can see, there is little difference in the usage. 

There is one small difference: I think it is more natural to specify the cost function reduction as a fraction instead of the expect fmin in the first iteration, so I introduced a "new" runtime parameter dfminFrac that is unknown to optim, and hence mitgcmuv_ad. Therefore I need to have different data.optim for mitgcmuv_ad and for my optim.x (see description here: https://github.com/mjlosch/optim_m1qn3). 

It would be trivial to include dfminFrac as a dummy parameter in mitgcmuv_ad, because within mitgcmuv_ad only the parameter "optimcycle" is used, as far as I can see (I removed all #include "optim.h" and the only recurring error message was a missing declaration of "optimcycle", except for one "fmin" in ctrl_pack.F, which could be easily avoided, I think, but again it's not really used). I never suggested this, because I had the feeling that this somewhat fragile construction should not be tempered with as long as it is used in ECCO runs. In fact, I'll do it now and you decide if you want to keep it.

dorugeber commented on doc/ocean_state_est/ocean_state_est.rst:
```diff
@@ -1239,6 +1239,70 @@ Error handling
+The code can be obtained at https://github.com/mjlosch/optim_m1qn3
+. The ``README`` contains short instructions how to build and use the
+code in combination with the ``tutorial_global_oce_optim``
+experiment. The usuage is very similar to the :filelink:`optim`
```
usuage -> usage

owang01 commented:
@gaelforget That is correct: the ECCO central production is using
http://wwwcvs.mitgcm.org/viewvc/MITgcm/MITgcm_contrib/ecco_utils/ecco_v4_release3_optimization/.

timothyas commented:
I've been using @mjlosch's offline m1qn3, I think the documentation would be nice!

jrscott commented:
This whole chapter of doc needs some work and another cleanup pass, but this contribution is helpful and LGTM.
